crates/polars-plan/src/dsl/mod.rs

```diff
@@ -561,11 +561,11 @@ impl Expr {
     /// It is the responsibility of the caller that the schema is correct by giving
     /// the correct output_type. If None given the output type of the input expr is used.
     ///
-    /// This difference with [map](Self::map) is that `apply` will create a separate `Series` per group.
+    /// This difference with [map_batches](Self::map_batches) is that `map_elements` will create a separate `Series` per group.
     ///
-    /// * `map` should be used for operations that are independent of groups, e.g. `multiply * 2`, or `raise to the power`
-    /// * `apply` should be used for operations that work on a group of data. e.g. `sum`, `count`, etc.
-    pub fn apply<F>(self, function: F, output_type: GetOutput) -> Self
+    /// * `map_batches` should be used for operations that are independent of groups, e.g. `multiply * 2`, or `raise to the power`
+    /// * `map_elements` should be used for operations that work on a group of data. e.g. `sum`, `count`, etc.
+    pub fn map_elements<F>(self, function: F, output_type: GetOutput) -> Self
```

Comment:
This one isn't the same as `map_elements` on the python side. UDF's on the rust side always get `Series`. The implementee is responsible for how to implement them.

The difference between `map` and `apply` is rather arbitrary and is more `map_batches` and `map_group_aware`.

You should always err to use `map_group_aware` (which I'd rather name just `map`).

Reply:
thanks, so
- map -> map_batches
- apply -> map
?

Reply:
I think silently renaming `apply` to `map` would be too confusing, and isn't worth doing

closing then, sorry for the noise, I should've thought this through better, will do so in the future

this isn't worth doing and isn't really necessary for https://github.com/pola-rs/polars/pull/13071, I'll note there that the Rust examples just show how to get the same output with the Rust API but aren't meant to be considered equivalent